packages/e2e-test-utils-playwright/src/admin/visit-site-editor.ts

```diff
@@ -37,8 +37,32 @@ export async function visitSiteEditor(
 		query.set( 'canvas', canvas );
 	}
 
+	const canvasLoader = this.page.locator(
+		// Spinner was used instead of the progress bar in an earlier version of
+		// the site editor.
+		'.edit-site-canvas-loader, .edit-site-canvas-spinner'
+	);
+
 	await this.visitAdminPage( 'site-editor.php', query.toString() );
 
+	// Try waiting for the canvas loader to appear first, so that the locator
+	// that waits for it to disappear doesn't resolve prematurely.
+	await canvasLoader.waitFor().catch( () => {} );
```

Comment:
What's the error we're expecting to catch here?

Reply:
In an unlikely case where the loader has already disappeared by the time we get to this line, it shouldn't fail the test – the action should continue executing. TBH, it hasn't happened to me so far, but it is possible so I think it's worth having this extra precaution as it doesn't really hurt us. Does that make sense?

Reply:
Makes sense

Reply:
@ellatrix, I think this line is the cause of regression. Now, each Site Editor > Navigation test takes around 2 minutes to run locally, instead of 5 seconds when I comment out this line. Running the tests in UI Mode gives a better breakdown of locator timers.

@WunderBart, I also realized that the util expects every site editor page it visits to have a `canvasLoader`, which isn't true. The DataView routes have no loaders for the moment.

Reply:
For reference: https://github.com/WordPress/gutenberg/pull/61816